be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -1409,6 +1409,38 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
     MonotonicStopWatch sw;
     sw.start();
 
+#ifdef USE_STAROS
+    // only used for CACHE SELECT, do not form any chunk to save CPU time,
+    // just read file content in `_scan_range`
+    if (_opts.lake_io_opts.cache_file_only) {
+        // read every column in this segment at once, maybe optimize this later
+        size_t buf_size = config::starlet_fs_stream_buffer_size_bytes;
+        if (buf_size <= 0) {
+            buf_size = 1048576; // 1MB
+        }
+        for (auto& [cid, stream] : _column_files) {
+            ASSIGN_OR_RETURN(auto vec, _column_iterators[cid]->get_io_range_vec(_scan_range));
+            std::unique_ptr<char[]> buf(new char[buf_size]);
```

Comment:
move this line out of the loop, no need to new the large array in every iteration.

Reply:
be done in next pr